### PR TITLE
Make network support optional

### LINKED ISSUE #64: Allow --update from a local tarball; allow disabling update warnings
Two simple changes would make tealdeer more offline-use friendly. In particular, when packaging it for use in an offline environment, one could include tldr's tarball (master.tar.gz) in the package itself, and make it run `tldr --update <tarball>` when installing.

Related: it would be nice to be able to disable "Cache wasn't updated for more than 30 days" warning (maybe at build time) -- if you're in an offline environment, you can't really update it so it's just an annoyance...

### PR DESCRIPTION
**Pull request by dbrgn**
Work in progress.

Fixes #64.

### CODE REVIEW

**Comment by dbrgn** [comment]
Currently blocked on https://github.com/assert-rs/assert_cmd/issues/64

File: src/cache.rs [hunk 1]
```diff
@@ -74,22 +94,38 @@ impl Cache {
...
         let mut buf: Vec<u8> = vec![];
         let bytes_downloaded = resp.copy_to(&mut buf)?;
         debug!("{} bytes downloaded", bytes_downloaded);
         Ok(buf)
     }
 
+    /// Download the archive from the network or load it from a file.
+    #[cfg(feature = "networking")]
+    fn load(&self) -> Result<Vec<u8>, TealdeerError> {
+        if self.source.starts_with("http") {
```
Comment (aldanor) [inline src/cache.rs]: Nitpicking, maybe something less stringly-typed? 😄 

```rust
pub enum Source {
    File(PathBuf),
    #[cfg(feature = "networking")]
    URL(String),
}
```

Comment (dbrgn) [reply src/cache.rs]: Hm, good point. Then we can move the parsing up in the hierarchy and convert the input string right after argument parsing :slightly_smiling_face:

**Comment by dbrgn** [comment]
I think this should be done (except the edge cases commented above). Thorough testing would be welcome :slightly_smiling_face:

**Comment by dbrgn** [comment]
@aldanor Would be great if you could quickly glance over the diff and test the branch!

**Comment by dbrgn** [comment]
Closing this, since added complexity without a real need is not worth it.

**Comment by dbrgn** [comment]
@yorkerhazan the repository should be cloned into `~/.cache/tealdeer/tldr-pages/`. If everything is set up correctly, then the file at `~/.cache/tealdeer/tldr-pages/index.json` and the directory at `~/.cache/tealdeer/tldr-pages/pages/` should exist.
